### ISSUE TITLE
False positive on when using a method on a variable

### ISSUE THREAD

**Issue by b4ldr** [rodjek/puppet-lint#937]
I noticed the following code triggers `WARNING: unquoted resource title on line 3`

```puppet
$foo = '/tmp/file.conf'
notice($foo.dirname)
file{$foo.dirname:}
```

```console
$ puppet apply ~/puppet_tests/puppet-lint_issue.pp                                                   
Notice: Scope(Class[main]): /tmp
Notice: Compiled catalog for desktop.home.arpa in environment production in 0.01 seconds
Notice: Applied catalog in 0.01 seconds
$ /usr/local/bin/puppet-lint -v  
puppet-lint 2.4.2
$ /usr/local/bin/puppet-lint ~/puppet_tests/puppet-lint_issue.p
WARNING: unquoted resource title on line 3
```
